Automattic/jetpack
# Add License Activation Flow to Jetpack Plugin

#### Changes proposed in this Pull Request:


* Add `js-package/licensing` to Jetpack Plugin
* Add `js-package/api` to `js-package/licensing` and hook up activation flow.

**Note** The activation flow is functionally complete but will still need additional styling in a follow-up PR
**Note** This PR does not contain "success" screens for every Jetpack Product. Those will be added in #21770 

Un-activated                       |  Activation Success
:-------------------------:|:-------------------------:
<img width="1075" alt="Screen Shot 2021-11-16 at 4 45 54 PM" src="https://user-images.githubusercontent.com/2810519/142115134-778ad135-c8dd-498d-b2de-08627efb9210.png"> | <img width="1098" alt="Screen Shot 2021-11-16 at 4 30 51 PM" src="https://user-images.githubusercontent.com/2810519/142115192-edc013ba-eff9-44d8-89b8-66f12b4d2901.png">


#### Jetpack product discussion



p1HpG7-cSg-p2

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






1. Navigate to `/wp-admin/admin.php?page=jetpack#/license/activation`
2. Put an invalid license into the license field and click "Activate"
3. Verify an error appears underneath the license field
4. Aquire a valid license ( complete a siteless Jetpack purchase with `add_filter( 'jetpack_license_checkout_m2_enabled', '__return_true' );` included in your `0-sandbox.php` and `public-api.wordpress.com` )
5. Put a valid license into the license field and click "Activate"
6. Verify you are taken to the success screen
